pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -244,16 +279,16 @@ private void rewriteRawMVForwardIndex(String column, ColumnMetadata existingColM
               .forForwardIndex(newCompressionType, _indexLoadingConfig.getColumnProperties());
 
       try (ForwardIndexCreator creator = indexCreatorProvider.newForwardIndexCreator(context)) {
-        // If creator stored type and the reader stored type do not match, throw an exception.
         if (!reader.getStoredType().equals(creator.getValueType())) {
+          // Creator stored type should match reader stored type for raw columns. We do not support changing datatypes.
           String failureMsg =
               "Unsupported operation to change datatype for column=" + column + " from " + reader.getStoredType()
                   .toString() + " to " + creator.getValueType().toString();
           throw new UnsupportedOperationException(failureMsg);
         }
```

Comment:
For dict enabled columns, this validation is not right. 
Ofcourse, I could check for dictCreator == null before performing the validation. But decided to extract it out as it would be more cleaner.